### PR TITLE
Migrate Validation/DTRecHit clients to DQMEDHarvester + update efficiency plots

### PR DESCRIPTION
**Pull request by battibass**
This PR follows up on the migration of the DQM clients of Validation/DTRecHit not touched in [PR 22082](https://github.com/cms-sw/cmssw/pull/22082) (from @fwyzard).

Includes:
- new classes to handle efficiencies harvesting in `plugins/Histograms.h`
- migration of the existing clients to `DQMEDHarvester`
- moving the computation of efficiency plots into clients
- extension of the `DTRecHitClients_cfi` configuration

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by fabiocos** [comment]
+1

File: Validation/DTRecHits/python/DTRecHitClients_cfi.py [hunk 1]
```diff
@@ -1,8 +1,22 @@
 import FWCore.ParameterSet.Config as cms
 
-dtrechitclients = cms.EDAnalyzer("DTRecHitClients")
-dt2dsegmentclients = cms.EDAnalyzer("DT2DSegmentClients")
-dt4dsegmentclients = cms.EDAnalyzer("DT4DSegmentClients")
+dtrechitclients = cms.EDProducer("DTRecHitClients",
```
Comment (schneiml) [inline Validation/DTRecHits/python/DTRecHitClients_cfi.py]: Note: these should go via https://github.com/cms-sw/cmssw/blob/master/DQMServices/Core/python/DQMEDHarvester.py . Not a functional problem for now, and likely not for the future, but a matter of keeping some order in the plugin types.

**Pull request merged**